### PR TITLE
feat: guest user lobbies

### PR DESCRIPTION
**Pull request by jimsynz**

### CODE REVIEW

File: lib/playdate/games/battle_wordle/dictionary/wordle-answers-alphabetical.txt [hunk 1]
```diff
@@ -1143,7 +1143,7 @@ livid
 llama
 loamy
 loath
-lobby
+AppInstance
```
Comment (juddey) [inline lib/playdate/games/battle_wordle/dictionary/wordle-answers-alphabetical.txt:1146]: Love the ol' find and replace whupsie :)

Comment (rellen) [inline lib/playdate/games/battle_wordle/dictionary/wordle-answers-alphabetical.txt:1146]: please revert this
search and replace FTW!

File: lib/playdate/apps/app_instance.ex [hunk 2]
```diff
@@ -29,11 +31,9 @@ defmodule Playdate.Apps.AppInstance do
   end
 
   attributes do
-    uuid_primary_key :id
-
-    attribute :room_code, :string, allow_nil?: false
-    attribute :app_users, {:array, Playdate.Apps.AppUser}
-    attribute :state, :string, allow_nil?: false, default: "idle"
+    attribute :code, :string, allow_nil?: false, primary_key?: true
+    attribute :app_users, {:array, Playdate.Apps.AppUser}, default: []
```
Comment (mbuhot) [inline lib/playdate/apps/app_instance.ex:35]: I think we can simplify the code by replacing this `app_users` array with a presence topic, and a calculation that will list the users via https://hexdocs.pm/phoenix/Phoenix.Presence.html#c:list/1

The `JoinAppInstanceChange` would need to accept a `pid` argument (usually the players LiveView pid) and call  https://hexdocs.pm/phoenix/Phoenix.Presence.html#c:track/4 to add the user to the instance.

As soon as a player closes their browser, they will be automatically removed from the instance thanks to presence.

File: lib/playdate/apps/app_user.ex [hunk 3]
```diff
@@ -6,17 +6,19 @@ defmodule Playdate.Apps.AppUser do
...
+      accept [:nickname]
     end
   end
 
   attributes do
-    attribute :email, :string, allow_nil?: false
-    attribute :user_id, :string, allow_nil?: false
+    uuid_primary_key :id
+    attribute :session_id, :uuid, allow_nil?: false
+    attribute :user_id, :uuid
```
Comment (mbuhot) [inline lib/playdate/apps/app_user.ex:21]: Is this field ever used? I can imagine we might have some rules like only registered users can start apps. But once the app is started, do we ever need to associate a user back to their email address?

File: lib/playdate_web/plugs/unique_session_identifier.ex [hunk 4]
```diff
@@ -0,0 +1,33 @@
+defmodule PlaydateWeb.UniqueSessionIdentifier do
```
Comment (mbuhot) [inline lib/playdate_web/plugs/unique_session_identifier.ex:1]: I like this pattern of co-locating a function plug with a LiveView on_mount hook to propagate data from the session to the socket assigns 👍

File: lib/playdate/apps/app_user.ex (hunk 3 above)
Comment (jimsynz) [reply lib/playdate/apps/app_user.ex:21]: I expect that we'll need the `user_id` to record scores?

File: lib/playdate/apps/app_instance.ex (hunk 2 above)
Comment (jimsynz) [reply lib/playdate/apps/app_instance.ex:35]: I was thinking about this, but we also need to be tolerant of people being unconnected for a few seconds. I was thinking that a short timeout after presence says that they've left may be the way go to?

File: lib/playdate/games/battle_wordle/dictionary/wordle-answers-alphabetical.txt (hunk 1 above)
Comment (jimsynz) [reply lib/playdate/games/battle_wordle/dictionary/wordle-answers-alphabetical.txt]: d'oh!

**Pull request merged**